persistenceOne/persistenceCore
# Anmol/release static

## 1. Overview

Create static release binaries in actions via building in docker.
Combine build and publish into one step.
Inspiration: https://github.com/osmosis-labs/osmosis/blob/main/.github/workflows/release.yml

## 2. Implementation details
* Create `Dockerfile.release` which is used for release
* Move dockerfiles in `docker` dir
* Add make commands for release yaml
* Rename `deploy.yml` to `release.yml`
* Build based on platform, instead of having 2 commands for 2 platforms, we give `PLATFORM` as variable

## 3. How to test/use
* Run `make release-build-platform PLATFORM=amd64 VERSION=test`
* Run `make release-build-platform PLATFORM=arm64 VERSION=test`
* Run `make release-git release-sha`
* Check the `release/` dir

## 4. Checklist
- [ ] Does the Readme need to be updated?

## 5. Limitations (optional)
* For buiilding multi-platform binaries, need to call the make command 2 times
* Additional `docker/Dockerfile.release` dockerfile, maybe we need to find a way to combine the 2

## 6. Future Work (optional)
* Combine docker files into more unified images
